### PR TITLE
Adapted Tom to new deptables location and layout

### PR DESCRIPTION
**Pull request by Lex-2008**
They moved from README.md file in deps-packaging subdir to README.md in the repo
root, and also titles include URLs now, too.

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a pull request! Maybe @olehermanse can review this?

**Comment by Lex-2008** [comment]
ready 2b reviewed!

**Comment by olehermanse** [comment]
@Lex-2008 could you run this and paste the output here in a comment?

**Comment by Lex-2008** [comment]
Text output:
```
[lex@flower2 cf-bottom]$ python -m tom -i -t cf-bottom -l info
<@cf-bottom> depstable: 3.12.x,3.15.x,master
fatal: remote upstream already exists.
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
Unpacking objects: 100% (13/13), 4.04 KiB | 229.00 KiB/s, done.
remote: Total 13 (delta 7), reused 7 (delta 7), pack-reused 5
From github.com:cfengine/buildscripts
 * branch            master     -> FETCH_HEAD
   d4d331c..5b55383  master     -> upstream/master
Switched to branch 'master'
Your branch is behind 'upstream/master' by 6 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
HEAD is now at 5b55383 Merge pull request #793 from olehermanse/mailmap
From github.com:cfengine/buildscripts
 * branch            3.12.x     -> FETCH_HEAD
Switched to branch '3.12.x'
Your branch is up to date with 'upstream/3.12.x'.
HEAD is now at 8331bcc Merge pull request #784 from Lex-2008/ENT-5363-suse15-3.12.x
Already up to date.
From github.com:cfengine/buildscripts
 * branch            3.15.x     -> FETCH_HEAD
Switched to branch '3.15.x'
Your branch is up to date with 'upstream/3.15.x'.
HEAD is now at 7f335cf Merge pull request #783 from Lex-2008/ENT-5363-suse15-3.15.x
Already up to date.
From github.com:cfengine/buildscripts
 * branch            master     -> FETCH_HEAD
Switched to branch 'master'
Your branch is up to date with 'upstream/master'.
HEAD is now at 5b55383 Merge pull request #793 from olehermanse/mailmap
Already up to date.
From github.com:cfengine/buildscripts
 * branch            master     -> FETCH_HEAD
Already on 'master'
Your branch is up to date with 'upstream/master'.
HEAD is now at 5b55383 Merge pull request #793 from olehermanse/mailmap
[WARNING] didn't find dep in line [| lcov             | 1.14   | 1.14   | 1.14   |]
[WARNING] didn't find dep in line [| git              |        |        |        |]
[WARNING] didn't find dep in line [| rsync            |        |        |        |]
[WARNING] didn't find dep in line [| libgcc                                                                           |        |        |        | AIX and Solaris only     |]
Switched to a new branch 'deptables-2021-01-13_14_03_25_765604'
[deptables-2021-01-13_14_03_25_765604 236620b] Update dependency tables
Enumerating objects: 1, done.
Counting objects: 100% (1/1), done.
Writing objects: 100% (1/1), 596 bytes | 99.00 KiB/s, done.
Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
remote: 
remote: Create a pull request for 'deptables-2021-01-13_14_03_25_765604' on GitHub by visiting:
remote:      https://github.com/Lex-2008/buildscripts/pull/new/deptables-2021-01-13_14_03_25_765604
remote: 
To github.com:Lex-2008/buildscripts.git
 * [new branch]      deptables-2021-01-13_14_03_25_765604 -> deptables-2021-01-13_14_03_25_765604
[INFO] SLACK: <@con>: Dependency tables:
PR: https://github.com/cfengine/buildscripts/pull/794
<@cf-bottom> 
```
Note that created PR doesn't change anything, since tables were already updated by Tom in https://github.com/cfengine/buildscripts/pull/790

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6014/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6014/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6014/

File: tom/dependencies.py [hunk 1]
```diff
@@ -316,33 +316,53 @@ def update_deps_version(self, branches):
...
             if line.startswith('| CFEngine version '):
                 has_notes = 'Notes' in line
-                branches_text = ' | '.join([branch.ljust(branch_column_widths[branch]) for branch in branches])
-                line = '| CFEngine version | ' + branches_text + (' | Notes |' if has_notes else ' |')
-            elif line.startswith('|  --'):
-                branches_text = ' | '.join(['-'*branch_column_widths[branch] for branch in branches])
-                line = '|  --------------  | ' + branches_text + (' | ----- |' if has_notes else ' |')
+                # Desired output row: ['CFEngine version', branches..., 'Notes']
+                # Also note that list addition is concatenation: [1] + [2] == [1, 2]
+                row = ( ["CFEngine version"] +
```
Comment (vpodzime) [inline tom/dependencies.py:321]: Please drop the spaces after and before parentheses.

File: tom/dependencies.py [hunk 2]
```diff
@@ -316,33 +316,53 @@ def update_deps_version(self, branches):
...
+                                  ( [column_widths[-2]-2] if has_notes else [] ) # "Notes", if exists
+                                 )
+                line = '| ' + (" | ".join((val.ljust(width) for val, width in zip(row, column_widths)))) + ' |'
+            elif line.startswith('| --'):
+                line = '| ' + (' | '.join(('-'*width for width in column_widths))) + ' |'
             else:
-                dep = re.match('\\|  ([a-z0-9-]*) ', line, flags=re.IGNORECASE)
-                if dep:
-                    dep = dep.group(1)
+                match = re.match('\| \[([a-z0-9-]*)\]\((.*?)\) ', line, flags=re.IGNORECASE)
```
Comment (vpodzime) [inline tom/dependencies.py:341]: A matching example would be nice here.

**Review by vpodzime — COMMENTED** [review]
```
[WARNING] didn't find dep in line [| lcov             | 1.14   | 1.14   | 1.14   |]
```
this line looks correct and should be parsed, right?

**Comment by Lex-2008** [comment]
> ```
> [WARNING] didn't find dep in line [| lcov             | 1.14   | 1.14   | 1.14   |]
> ```
> 
> this line looks correct and should be parsed, right?

Not really. All other dependencies always have URL in them, while these three come from [Build dependencies](https://github.com/cfengine/buildscripts/blob/master/README.md#build-dependencies) table and don't have URLs.

**Comment by olehermanse** [comment]
> Not really. All other dependencies always have URL in them, while these three come from Build dependencies table and don't have URLs.

@Lex-2008 So they are correct. You are contradicting yourself in the same sentence ;)

**Review by olehermanse — CHANGES_REQUESTED** [review]
The commit message (and PR title) should explain what this changes.

Tom built deptables before, you updated it, Tom still builds deptables. What is the difference?

**Review by olehermanse — APPROVED** [review]
Looks good to me now. All the regex is quite cryptic to me, but thanks for the examples and comments.

**Review by cf-bottom — APPROVED** [review]
I trust @olehermanse, approved!

File: tom/dependencies.py [hunk 3]
```diff
@@ -316,33 +316,58 @@ def update_deps_version(self, branches):
...
-                branches_text = ' | '.join(deps_table[dep][branch].ljust(branch_column_widths[branch]) for branch in branches)
+                        note = ''
+                    else:
+                        note = note.group(1)
                 if in_hub:
                     dep = re.sub('-hub$', '', dep)
-                line = '|  %-15s | %s %s' % (dep, branches_text, (note if has_notes else '|'))
+                row = ( ["[%s](%s)" % (dep_title, url)] +
+                        [deps_table[dep][branch] for branch in branches] +
+                        ( [note] if has_notes else [] )
```
Comment (vpodzime) [inline tom/dependencies.py:368]: Some more unnecessary spaces inside parentheses.

**Review by vpodzime — APPROVED** [review]
ACK

**Pull request merged**